datacube.py

```diff
@@ -0,0 +1,382 @@
+    - s1_items (list): List of Sentinel-1 items.
+    - dem_items (list): List of DEM items.
+    - BBOX (tuple): Bounding box coordinates in the format (minx, miny, maxx, maxy).
+    - resolution (int): Spatial resolution.
+    - epsg (int): EPSG code for the coordinate reference system.
+
+    Returns:
+    - tuple: A tuple containing xarray DataArrays for Sentinel-2, Sentinel-1, and DEM.
+    """
+    da_sen2: xr.DataArray = stackstac.stack(
+        items=s2_items[0],
+        epsg=26910,  # UTM Zone 10N
+        assets=S2_BANDS,
+        bounds_latlon=BBOX,  # W, S, E, N
+        resolution=10,  # Spatial resolution of 10 metres
+        xy_coords="center",  # pixel centroid coords instead of topleft corner
+        dtype=np.float32,
+        fill_value=np.nan,
+    )
+
+    da_sen1: xr.DataArray = stackstac.stack(
+        items=s1_items[1:],  # To only accept the same orbit state and date. Need better way to do this.
+        assets=["vh", "vv"],  # SAR polarizations
+        epsg=26910,  # UTM Zone 10N
+        bounds_latlon=BBOX,  # W, S, E, N
+        xy_coords="center",  # pixel centroid coords instead of topleft corner
+        dtype=np.float32,
+        fill_value=np.nan,
+    )
+
+    # To fix TypeError: Invalid value for attr 'spec'
+    da_sen1.attrs["spec"] = str(da_sen1.spec)
+
+    # To fix ValueError: unable to infer dtype on variable None
+    for key, val in da_sen1.coords.items():
+        if val.dtype == "object":
+            print("Deleting", key)
+            da_sen1 = da_sen1.drop_vars(names=key)
+
+    # Create xarray.Dataset datacube with VH and VV channels from SAR
+    da_vh: xr.DataArray = da_sen1.sel(band="vh", drop=True).rename("vh")
+    da_vv: xr.DataArray = da_sen1.sel(band="vv", drop=True).rename("vv")
+    ds_sen1: xr.Dataset = xr.merge(objects=[da_vh, da_vv], join="override")
+
+    # print(ds_sen1)
+
+    # da_sen1 = da_sen1.drop([da_sen1.time[0].values], dim='time') # Remove first scene which has ascending orbit
+
+    da_dem: xr.DataArray = stackstac.stack(
+        items=dem_items,
+        epsg=26910,  # UTM Zone 10N
+        bounds_latlon=BBOX,  # W, S, E, N
+        resolution=10,  # Spatial resolution of 10 metres
+        xy_coords="center",  # pixel centroid coords instead of topleft corner
+        dtype=np.float32,
+        fill_value=np.nan,
+    )
+
+    _, index = np.unique(da_dem['time'], return_index=True)  # Remove redundant time
+    da_dem = da_dem.isel(time=index)
+
+    return da_sen2, da_sen1, da_dem
+
+
+def merge_datarrays(da_sen2, da_sen1, da_dem):
+    """
+    Merge xarray DataArrays for Sentinel-2, Sentinel-1, and DEM.
+
+    Parameters:
+    - da_sen2 (xr.DataArray): xarray DataArray for Sentinel-2 data.
+    - da_sen1 (xr.DataArray): xarray DataArray for Sentinel-1 data.
+    - da_dem (xr.DataArray): xarray DataArray for DEM data.
+
+    Returns:
+    - xr.DataArray: Merged xarray DataArray.
+    """
+    da_merge = xr.merge([da_sen2, da_sen1, da_dem], compat='override')
```

Comment:
We can then probably use `concat` to merge all data arrays along the band dimensions.

Reply:
Think that's already handled by `xr.merge`. It returns a:
```
Merged datarray:  <xarray.Dataset>
Dimensions: (time: 1, band: 14, x: 11365, y: 11363)
```